azure-common/azure/profiles/multiapiclient.py

```diff
@@ -0,0 +1,81 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+from . import KnownProfiles, ProfileDefinition
+
+class InvalidMultiApiClientError(Exception):
+    """If the mixin is not used with a compatible class.
+    """
+    pass
+
+class MultiApiClientMixin(object):
+    """Mixin that contains multi-api version profile management.
+
+    To use this mixin, a client must define two class attributes:
+    - LATEST_PROFILE : a ProfileDefinition correspond to latest profile
+    - _PROFILE_TAG : a tag that filter a full profile for this particular client
+
+    This should not be used directly and will only provide private methods.
+    """
+
+    def __init__(self, api_version=None, profile=KnownProfiles.default, **kwargs):
+
+        try:
+            type(self).LATEST_PROFILE
+        except AttributeError:
+            raise InvalidMultiApiClientError("To use this mixin, main client MUST define LATEST_PROFILE class attribute")
+
+        try:
+            type(self)._PROFILE_TAG
+        except AttributeError:
+            raise InvalidMultiApiClientError("To use this mixin, main client MUST define _PROFILE_TAG class attribute")
+
+        if api_version and profile is not KnownProfiles.default:
+            raise ValueError("Cannot use api-version and profile parameters at the same time")
+
+        if api_version:
+            self.profile = ProfileDefinition({
+                self._PROFILE_TAG: {
+                    None: api_version
+                }},
+                self._PROFILE_TAG + " " + api_version
+            )
+        elif isinstance(profile, dict):
+            self.profile = ProfileDefinition({
+                self._PROFILE_TAG: profile,
+                },
+                self._PROFILE_TAG + " dict"
```

Comment:
Is this just so you can set a default value?

Reply:
I already support a dict as parameter, where key are operation group and value are api-version. I migrate that dict to the new internal format.